cli/pull.ml

```diff
@@ -89,7 +89,7 @@ let pull ~duniverse_dir src_dep =
   Bos.OS.Dir.delete ~recurse:true output_dir >>= fun () ->
   Exec.git_shallow_clone ~output_dir ~remote:upstream ~ref () >>= fun () ->
   Exec.git_rev_parse ~repo:output_dir ~ref:"HEAD" () >>= fun head_commit ->
-  checkout_if_needed ~head_commit ~output_dir ~ref ~commit () >>= fun () ->
+  checkout_if_needed ~head_commit ~output_dir ~ref ~commit ~dir () >>= fun () ->
```

Comment:
To avoid passing an unneeded argument, you could use ``|> R.reword_error (fun (`Msg _) -> `Commit_is_gone dir)`` here.